File: src/codility/java/counting/MissingInteger.java
package codility.java.counting;

import java.util.HashSet;

public class MissingInteger {
    public int solution(int[] A) {
        HashSet<Integer> set = new HashSet<>();

        for (int a : A) {
            set.add(a);
        }
        int ans = -1;
        for (int i = 1; i<= A.length +1;i++) {
            if (!set.contains(i)) {
                ans = i;
                break;
            }
        }

        return ans;
    }

}

File: src/codility/kotlin/MissingInteger.kt
package codility.kotlin

class MissingInteger {

    fun solution(A: IntArray): Int {
        val set = A.toHashSet()

        var result = 1
        while(true) {
            if (!set.contains(result)) {
                break;
            }
            result++
        }
        return result
    }
}